crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -1160,4 +1160,28 @@ mod tests {
         world.flush();
         assert_eq!(vec!["event", "event"], world.resource::<Order>().0);
     }
+
+    #[test]
+    fn observer_on_remove_during_despawn_spawn_empty() {
```

Comment:
A comment about how prior to this PR this would panic would be helpful to motivate the test since there's no asserts.

Reply:
Done!